file.go

```diff
@@ -838,7 +841,27 @@ func (f *FilePages) ReadPage() (Page, error) {
 
 		f.index++
 		if f.skip == 0 {
-			return page, nil
+			// f.skip==0 can be true:
+			//  (1) while reading a row of a column which has multiple values (ie. X.list.element) and values continue
+			//  across pages. In that case we just want to keep reading without skipping any values.
+			//  (2) when seeking to a specific row and trying to reach the start offset of the first
+			//  row in a new page.
+			if !seekToRowStart {
+				// keep reading values from beginning of new page
+				return page, nil
+			} else {
+				if page.NumRows() == 0 {
+					// if current page does not have any rows, continue until a page with at least 1 row is reached
+					Release(page)
+					continue
+				}
+				// In V1 data pages, new page does not necessarily start with a new row.
```

Comment:
is this a v1 possibility only? page.Slice() can be an expensive operation b/c (depending on the page type) it can loop through all repetition levels to find row 0 and row page.NumRow().

if it's v1 only should we test for that before slicing?

Reply:
Yes, that is my understanding. [v2 data pages](https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L725-L730) always start at a row boundary.

Based on this, I have restructured the if-else so it's hopefully easier to follow.

Reply:
could we use the repetitionlevels in the page? something like this (except with more bounds checking) would be a direct check on the actual situation that we're correcting for. 

```
// this page doesn't start on a row boundary and we're seeking
if seekToRowStart && page.RepetitionLevels()[0] != 0 {
}
```

i think this would work

Reply:
thanks, I added the repetition level check to avoid the page slice if the page starts at row boundary [here](https://github.com/parquet-go/parquet-go/pull/277/commits/99e961f01ca0a311a696bbf5742723e3d961b19b). At this point, `seekToRowStart` is known to be `true` so no need to check that too.